src/Microsoft.DotNet.Darc/DarcLib/Helpers/DependencyFileManager.cs

```diff
@@ -836,7 +892,8 @@ private static XmlComment GetFirstMatchingComment(XmlNode nodeToCheck, string co
     private async Task AddDependencyToVersionDetailsAsync(
         string repo,
         string branch,
-        DependencyDetail dependency)
+        DependencyDetail dependency,
+        bool repoHasVersionDetailsProps)
```

Comment:
One issue with adding/removing individual dependencies like this is that we always have to load the xml from git, do our operation, and then regenerate the VersionDetailsProps. We should have something like `RemoveDependenciesAsync` that would only load the files once and only generate `Version.Details.props` once. I kind of feel like it'd be better to that in a separate PR